mesonbuild/backend/backends.py

```diff
@@ -1303,7 +1303,7 @@ def get_regen_filelist(self) -> T.List[str]:
         '''List of all files whose alteration means that the build
         definition needs to be regenerated.'''
         deps = OrderedSet([str(Path(self.build_to_src) / df)
-                           for df in self.interpreter.get_build_def_files()])
+                           for df in self.build.def_files])
```

Comment:
I don't think the intermediate list is needed here. I would rather use a generative expression.

Reply:
I was doing the minimal change (also, in general `foo(x for y in z)` is slower than `foo([x for y in z])` even though the latter is apparently less efficient).